tests/functional/graphing/12-unsatisfiable.t

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test un-satisfiable dependencies are correctly identified in graphs
+. "$(dirname "$0")/test_header"
+#-------------------------------------------------------------------------------
+set_test_number 9
+init_workflow "${TEST_NAME_BASE}" <<'__FLOW_CONFIG__'
+[scheduler]
+    allow implicit tasks = True
+[scheduling]
+    cycling mode = integer
+    initial cycle point = 1
+    [[graph]]
+        # two un-satisfiable dependencies
+        # (the inter-cycle deps are missaligned with the recurrences)
+        1/P2 = a[-P2] => b
+        2/P2 = b[-P2] => a
+__FLOW_CONFIG__
+
+#-------------------------------------------------------------------------------
+# cylc validate should log warnings for the un-satisfiable dependencies
+TEST_NAME="${TEST_NAME_BASE}-validate"
+run_ok "${TEST_NAME}" cylc validate "${WORKFLOW_NAME}"
+grep_ok "Detected un-satisfiable dependencies" "${TEST_NAME}.stderr"
+grep_ok "1/a => 3/b" "${TEST_NAME}.stderr"
+grep_ok "2/b => 4/a" "${TEST_NAME}.stderr"
+
+#-------------------------------------------------------------------------------
+# cylc graph should log warnings for the un-satisfiable dependencies
+TEST_NAME="${TEST_NAME_BASE}-graph"
```

Comment:
IMO this should be a separate test - you are checking the output as a dot file. 

I presume that the other two output formats in the `--help` docs are potentially dependent on the environment, and that tests would realistically be limited to checking that the files are a png/svg. If this isn't the case can we have a test, especially for the SVG, which is after all a text file.

Reply:
It's checking the same functionality against the same config so I think these two should be in together.

I haven't tested formats like SVG and PNG because that conversion is done by graphviz and testing graphviz output feels out of scope.